### DISCUSSION TITLE
Deploying TurboRepo Next.js Project with Git Submodule Private Repository on Vercel: Issue with Visibility

### DISCUSSION THREAD

**Discussion by tejaslende** [vercel/turborepo#7857] [Help] [answered]
### Summary

In this discussion, we'll address a common issue encountered when deploying TurboRepo Next.js projects on Vercel that incorporate Git submodule private repositories. Despite successful deployment, the expected submodule content fails to display on the deployed site.

We'll explore potential causes for this visibility issue, such as authentication challenges with private repositories, submodule configurations, and Vercel deployment settings. Additionally, we'll discuss strategies for resolving this problem, including adjusting submodule permissions, ensuring proper authentication methods, and configuring Vercel deployment settings to accommodate submodule dependencies.

By sharing experiences, troubleshooting steps, and potential solutions, participants can collaborate to overcome challenges associated with deploying TurboRepo Next.js projects utilizing Git submodule private repositories on the Vercel platform.

**Comment by anthonyshew** [comment, marked as answer]
Vercel does not support git submodules. You may have better luck discussing at https://github.com/vercel/vercel/discussions

**Reply by tejaslende** [reply]
Can I deploy on AWS Amplify?
